packages/compiler/src/compiler/compile.ts

```diff
@@ -158,35 +159,80 @@ export class Compile {
     return sourceRef
   }
 
-  private addStrayText(text: string, node: TemplateNode) {
+  private addStrayText(text: string, node?: TemplateNode) {
     const sourceRef = this.getSourceRef(text, node)
 
     this.accumulatedText.text += text
     if (sourceRef) this.accumulatedText.sourceMap.push(sourceRef)
   }
 
+  // We should find another way to ensure SourceRefs
+  // are in sync, this seems like a hack
+  private outdentSourceRefs(
+    text: string,
+    sourceMap: PromptlSourceRef[],
+  ): PromptlSourceRef[] {
+    const indent = getCommonIndent(text)
+    let position = 0
+    text = text
+      .split('\n')
+      .map((line) => {
+        const offset = line.length - line.slice(indent).length
+        line = line.slice(indent)
+        sourceMap = sourceMap.map((ref) => ({
+          ...ref,
+          start: ref.start >= position ? ref.start - offset : ref.start,
+          end: ref.end >= position ? ref.end - offset : ref.end,
+        }))
+        position += line.length + 1
+        return line
+      })
+      .join('\n')
+
+    const offset = text.length - text.trimStart().length
+    text = text.trimStart()
+    sourceMap = sourceMap.map((ref) => ({
+      ...ref,
+      start: ref.start - offset,
+      end: ref.end - offset,
+    }))
+
+    text = text.trimEnd()
+    sourceMap = sourceMap.map((ref) => ({
+      ...ref,
+      start: Math.min(ref.start, text.length),
+      end: Math.min(ref.end, text.length),
+    }))
+
+    return sourceMap
+  }
```

Comment:
is this to support collapsing variables in old compiler? Honestly... i wouldn't have done it 😂 I guess I'm too late for that